src/redux/initiative-tracker/initiativeTrackerSlice.ts

```diff
@@ -27,12 +53,11 @@ export const initiativeTrackerSlice = createSlice({
   name: "initiativeTracker",
   initialState,
   reducers: {
-    addCreature: (state, action: PayloadAction<InitiativeCreature>) => {
-      action.payload.uuid = uuidv4();
-
+    addCreature: (state, action: PayloadAction<InitiativeCreatureExternal>) => {
       const currentTurnUuid = state.creatures[state.currentTurn]?.uuid;
 
-      state.creatures.push(action.payload);
+      const creature: InitiativeCreature = { order: 0, uuid: uuidv4(), ...action.payload };
+      state.creatures.push(creature);
       sortInitiativeCreatures(state.creatures);
 
       // If combat has started, preserve the turn of the creature whose turn it currently is
```

Comment:
Why is this comment here. Is this a todo?

Reply:
clarifies the logic below it